### PR TITLE
CLDC-2014 remove text field for buyer 1 nationality other

### PR DESCRIPTION
**Pull request by arfacamble**
https://digital.dclg.gov.uk/jira/browse/CLDC-2014

image

### CODE REVIEW

File: spec/models/form/sales/pages/nationality1_spec.rb [hunk 1]
```diff
@@ -1,22 +1,20 @@
...
   let(:subsection) { instance_double(Form::Subsection) }
 
   it "has correct subsection" do
-    expect(page.subsection).to eq(subsection)
+    expect(page.subsection).to be subsection
   end
 
   it "has correct questions" do
-    expect(page.questions.map(&:id)).to eq(%w[national othernational])
+    expect(page.questions.map(&:id)).to eq %w[national]
```
Comment (kosiakkatrina) [inline spec/models/form/sales/pages/nationality1_spec.rb:13]: Any reason for changing the syntax?

File: app/models/form/sales/pages/nationality1.rb [hunk 2]
```diff
@@ -7,15 +7,12 @@ def initialize(id, hsh, subsection)
...
     ]
   end
 
   def questions
-    @questions ||= [
-      Form::Sales::Questions::Nationality1.new(nil, nil, self),
-      Form::Sales::Questions::OtherNationality1.new(nil, nil, self),
-    ]
+    @questions ||= [Form::Sales::Questions::Nationality1.new(nil, nil, self)]
   end
```
Comment (kosiakkatrina) [inline app/models/form/sales/pages/nationality1.rb:17]: Are we meant to be removing it for both years?

File: spec/models/form/sales/questions/nationality1_spec.rb [hunk 3]
```diff
@@ -1,38 +1,36 @@
...
-  subject(:question) { described_class.new(question_id, question_definition, page) }
+  subject(:question) { described_class.new(nil, nil, page) }
 
-  let(:question_id) { nil }
-  let(:question_definition) { nil }
   let(:page) { instance_double(Form::Page) }
 
   it "has correct page" do
-    expect(question.page).to eq(page)
+    expect(question.page).to be page
```
Comment (kosiakkatrina) [inline spec/models/form/sales/questions/nationality1_spec.rb:9]: Do we need to change the syntax?

File: spec/models/form/sales/pages/nationality1_spec.rb (hunk 1 above)
Comment (arfacamble) [reply spec/models/form/sales/pages/nationality1_spec.rb:13]: you mean missing out the parentheses? Just less typing, equal or improved readability

File: app/models/form/sales/pages/nationality1.rb (hunk 2 above)
Comment (arfacamble) [reply app/models/form/sales/pages/nationality1.rb:17]: yes. clarified with Miguel

File: spec/models/form/sales/questions/nationality1_spec.rb (hunk 3 above)
Comment (arfacamble) [reply spec/models/form/sales/questions/nationality1_spec.rb:9]: be is more specific in this case, I think we would like it to be the exact same object.
lack of parentheses as above, less typing, equivalent or marginally better readability

File: spec/models/form/sales/pages/nationality1_spec.rb (hunk 1 above)
Comment (kosiakkatrina) [reply spec/models/form/sales/pages/nationality1_spec.rb:13]: I'd keep it consistent with the rest of the codebase, plus less changes when reviewing

**Review by kosiakkatrina — APPROVED** [review]

**Pull request merged**